openssl/src/x509/mod.rs

```diff
@@ -1670,6 +1671,27 @@ impl X509Revoked {
         X509Revoked,
         ffi::d2i_X509_REVOKED
     }
+
+    pub fn new(to_revoke: &X509) -> Result<Self, ErrorStack> {
```

Comment:
Hi,

wouldn't it make sense to use a "non-owned" reference for the certificate?

```suggestion
    pub fn new(to_revoke: &X509Ref) -> Result<Self, ErrorStack> {
```

And for the rest of your PR code as well.

Reply:
makes sense, thank you for the suggesting :)